### PR TITLE
Fix allow_ip6 issue in astats plugin

### PR DESCRIPTION
**Pull request by weifensh**
IPv6 address is 128bit/16byte long;
AF_INET to AF_INET6 for inet_pton calls;
some other changes.

### CODE REVIEW

**Comment by weifensh** [comment]
linking to issue #1557

File: traffic_server/plugins/astats_over_http/astats_over_http.c [hunk 1]
```diff
@@ -538,7 +538,7 @@ void TSPluginInit(int argc, const char *argv[]) {
 	info.support_email = "justin@fp-x.com";
 	astatsLoad = time(NULL);
 
-	if (TSPluginRegister(&info) != TS_SUCCESS) {
+	if (TSPluginRegister(TS_SDK_VERSION_3_0, &info) != TS_SUCCESS) {
```
Comment (PSUdaemon) [inline traffic_server/plugins/astats_over_http/astats_over_http.c]: Is this related to ipv6 changes?

**Comment by PSUdaemon** [comment]
I have concerns about this. The repo you are using for this is not the canonical one. I'd also like to obsolete this plugin in favor of the `stats_over_http` plugin from the main ATS repo and a new `remap_stats` like plugin that just adds the few stats this plugin has manually added. I'd be much more interested in seeing that work completed instead of further work on this one. Is this something you might be interested in helping out with?

**Comment by limited** [comment]
Thanks for the review Phil-
  What is the canonical repo for the astats plugin?

**Comment by PSUdaemon** [comment]
It's an internal Comcast one that is branched off of the `trafficserver` repo :). This is why I'd like to change all that. The `astats` plugin is heavily overlapping with `stats_over_http` and likely if we made a `system_stats` plugin much like the `remap_stats` plugin we could get it into the trafficserver mainline. It really doesn't belong in a Traffic-Control repo at all, imo.

**Comment by limited** [comment]
I agree `astats` is out of Traffic Control's scope and would be interested in seeing TC use `stats_over_http` instead. 

If we are re-examining module boundaries, does it still make sense for Traffic Server to report on system stats?  If we ever decide to monitor something other than bandwidth or load average, we'd need to come out with a new build of `ats`. Have you thought at all about using some other monitoring tool  to poll the system health instead of binding it to `astats`? 

In thinking about this, it could be useful in a few different cases:
1. If a cache other than `trafficserver` is used, we don't need to reimplement system stats gathering.
2. We haven't observed our Traffic Routers getting overloaded, but this could open the door for health monitoring and load balancing amongst TRs as well.

The downside is of course more packages to install, more configuration to push down and one more thing to break.

**Comment by PSUdaemon** [comment]
I think what we are both proposing is compatible. If we end up with a separate "system stats" plugin that added that stuff, then it can be swapped with any other sort of thing given appropriate support in traffic monitor. This would require more work, but it's feasible. I think what I am proposing is something that can be accomplished easier, as it's a slight renaming of variables. Then we can make it more generic down the road to meet your proposal's requirements. It would be more flexible and people could go as far as they wanted. For example, integrate with their already existing monitoring which means fewer packages instead of more.

**Comment by limited** [comment]
Yeah, I think this is probably step in the right direction. I like that its a more incremental change with some possibilities for future improvement. 

To use a new `system_stats` plugin, does TM make a separate HTTP call to the cache to get the system stats? Or will the system stats be included as part of the `stats` call?

**Comment by dneuman64** [comment]
I think we should get a new issue opened up for getting rid of astats in favor of something else...

**Comment by PSUdaemon** [comment]
It will register new plugin stats much like the `remap_stats` plugin does.

https://github.com/apache/trafficserver/blob/master/plugins/experimental/remap_stats/remap_stats.c#L45-L90

And those stats will end up in the output of `stats_over_http`. So there would be some "schedule every" type even handler that looks at the system interfaces like `astats` does and then update the internal stats that are then exposed via `stats_over_http`.

File: traffic_server/plugins/astats_over_http/astats_over_http.c (hunk 1 above)
Comment (weifensh) [inline traffic_server/plugins/astats_over_http/astats_over_http.c]: No, it's not related to ipv6. This change just helps astats to get compiled with ATS 5.3.2.

**Comment by dneuman64** [comment]
can this be closed since we are going to go a different direction?
